torch/utils/data/dataloader.py

```diff
@@ -45,7 +44,6 @@ def _worker_loop(dataset, index_queue, data_queue, collate_fn):
     while True:
         r = index_queue.get()
         if r is None:
-            data_queue.put(None)
             break
```

Comment:
For maneger_thread, `self.done_event' and manually sending `None` from main thread is more reliable than workers.